<a href="https://colab.research.google.com/github/solomeya12345-code/python_for_data_analitic/blob/main/Homework_1_1_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [43]:
a =101
b = 210
c = 111
d = 402
(a + b) * c / d

85.8731343283582

Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [44]:
11283 % 89

69

3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [45]:
11283 // 89

126

4. Запишіть у змінну `name` ваше імʼя.

In [5]:
name = "Solomiia"

5. Запишіть у змінну `surname` ваше прізвище.

In [6]:
surname = "Leskiv"

6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [7]:
name_surname = name + " " + surname
name_surname

'Solomiia Leskiv'

Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [46]:
name_surname.upper()

'SOLOMIIA LESKIV'

Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [9]:
name_surname[2]

'l'

Очікувана відповідь (тільки з вашими даними): 'n'

9. Виведіть першу та останню літеру вашого прізвища.

In [47]:
name_surname[0], name_surname[-1]

('S', 'v')

Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [30]:
len(surname)

6

Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [35]:
city = "Antananarivo"
print(city.lower().count("a"))

4


In [49]:
'Antananarivo'.lower().count('a')

4

Очікувана відповідь: 4

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [40]:
my_string = "The quick brown fox jumps over the lazy dog!!!"
print(my_string.startswith("The"))
print(my_string.endswith("dog"))

True
False


13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [51]:
words_list = my_string.split()
'_'.join(words_list)

'The_quick_brown_fox_jumps_over_the_lazy_dog!!!'

In [50]:
"_".join(my_string.split(" "))

'The_quick_brown_fox_jumps_over_the_lazy_dog!!!'

Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`